osmdroid-shape/src/main/java/org/osmdroid/shape/ShapeConverter.java

```diff
@@ -83,74 +91,74 @@
-                                polygon.setTitle(getSensibleTitle(polygon.getSnippet()));
-                            }
 
                             PointData[] points = aPolygon.getPointsOfPart(i);
                             List<GeoPoint> pts = new ArrayList<>();
 
                             for (PointData p : points) {
-                                GeoPoint pt = new GeoPoint(p.getY(), p.getX());
+                                GeoPoint pt = fixOutOfRange(new GeoPoint(p.getY(), p.getX()));
                                 pts.add(pt);
                             }
                             pts.add(pts.get(0));    //force the polygon to close
 
-                            polygon.setPoints(pts);
-                            final BoundingBox boundingBox = polygon.getBounds();
-                            polygon.setSubDescription(boundingBox.toString());
-                            folder.add(polygon);
+                            polygon.setPoints(pts); //points out of range should be fixed before this line.
+
+                            shapeMetaSetter.set(metadata, polygon);
+                            polygon.setSnippet(shapeMetaSetter.getSnippet());
+                            polygon.setTitle(shapeMetaSetter.getTitle());
 
+                            polygon.setSubDescription(shapeMetaSetter.getSubDescription());
 
+                            polygon.getFillPaint().set(shapeMetaSetter.getFillPaint());
+                            polygon.getOutlinePaint().set(shapeMetaSetter.getStrokePaint());
+                            folder.add(polygon);
                         }
                     }
-
                     break;
+
                     case POLYLINE: {
                         PolylineShape polylineShape = (PolylineShape) s;
                         for (int i = 0; i < polylineShape.getNumberOfParts(); i++) {
                             Polyline line = new Polyline(map);
-                            if (metadata != null) {
-                                metadata.setStringCharset(Charset.defaultCharset());
-                                line.setSnippet(metadata.toMap().toString());
-                                line.setTitle(getSensibleTitle(line.getSnippet()));
-                            }
+
                             PointData[] points = polylineShape.getPointsOfPart(i);
                             List<GeoPoint> pts = new ArrayList<>();
 
                             for (PointData p : points) {
-                                GeoPoint pt = new GeoPoint(p.getY(), p.getX());
+                                GeoPoint pt = fixOutOfRange(new GeoPoint(p.getY(), p.getX()));
                                 pts.add(pt);
                             }
-                            line.setPoints(pts);
+
+                            line.setPoints(pts);//points out of range should be fixed before this line.
+
+                            shapeMetaSetter.set(metadata, line);
```

Comment:
Just `shapeMetaSetter.set(metadata, line)`: the other three lines `line.setSnippet(shapeMetaSetter.getSnippet())`, `line.setTitle(shapeMetaSetter.getTitle())` and `line.setSubDescription(shapeMetaSetter.getSubDescription())` will be performed inside `shapeMetaSetter.set(metadata, line)`.